docs/source/authentication_with_modelon_impact.rst

```diff
@@ -0,0 +1,135 @@
+JupyterHub gateway authentication is necessary. From a notebook that is active within the MI 
+installation, one can create a connection to the external URL. In that case, this will be automatic
+based on the automatically populated ``JUPYTERHUB_API_TOKEN`` environment variable.
+
+An API key must be used in addition to the ``JUPYTERHUB API TOKEN`` to authenticate against the Modelon 
+Impact server operating inside the JupyterHub environment. If the client library is used within the
+JupyterHub environment, this is likewise handled automatically based on the key stored at 
+``/.impact/api.key`` in that environment.
+
+The python client could be easily initialized as follows for a user accessing the client library 
+in the JupyterHub environment:
+
+.. code-block:: pycon
+
+    >>>> from modelon.impact.client import Client
+    >>>> client = Client()
+
+
+JupyterHub authentication
+*************************
+
+**Note**:- This section is **ONLY** applicable for users using the client library used outside of
+the JupyterHub environment.
+
+If this client library is used outside of the environment, such as from your desktop, the JupyterHub API 
+token must be provided either interactively or in the environment variable ``JUPYTERHUB_API_TOKEN``. The token
+can be acquired using the token page at https://impact.modelon.cloud/hub/token or a similar page in the 
+on-premise installation. The below sections will cover details to generate an API key to log in to the 
+client.
+
+JupyterHub token generation
+###########################
+
+To create the JupyterHub token, go to the JupyterHub token manager (located at "https://impact.modelon.cloud/hub/token"). 
+You can get a secret token by choosing the "Request new API token" option. These are fully functional access tokens for 
+the JupyterHub API. Everything that can be done with JupyterHub can be done with these tokens.
+
+*Remember that this is the only time you will see the secret. Make sure to store it safely.*
+
+If the token is lost, you can always generate a new one by clicking the "Request new API token"
+button. While initializing the client, you will be asked to enter the JupyterHub API token in a prompt.
+
+.. code-block:: pycon
+
+    >>>> from modelon.impact.client import Client
+    >>>> client = Client(url="https://impact.modelon.cloud", interactive=True)
+    Enter JupyterHub API token:
+
+When the token has been entered the first time, it will be stored at located in the directory
+``.impact`` in the users home folder and used in future requests and the prompt will not be 
+shown again. 
+
+Another option is to save the token in the environmental variable ``JUPYTERHUB_API_TOKEN``, which 
+would allow the client to authenticate without engaging in any interaction prompts.
+
+To view a list of active tokens or revoke active tokens, the user can
+go to the JupyterHub token manager (located at "https://impact.modelon.cloud/token").

```

Comment:
I suggest using `https://impact.modelon.cloud/hub/token` everywhere for consistency (both work)